homework/unique_ptr/UniquePtr.hpp

```diff
@@ -0,0 +1,82 @@
+#pragma once
+#include "Exceptions.hpp"
+
+template <typename T>
+class UniquePtr
+{
+public:
+    UniquePtr() = default;
+    UniquePtr(T * rawPtr);
+    UniquePtr(UniquePtr<T> && otherPtr);
+    ~UniquePtr();
+
+    UniquePtr(const UniquePtr<T> &) = delete;
+    UniquePtr<T> & operator=(const UniquePtr<T> &) = delete;
+
+    UniquePtr<T> & operator=(UniquePtr<T> && otherUniquePtr);
+
+    T & operator*() const;
+    T * operator->() const;
+    T * get() const;
+    T * release();
+    void reset(T * newPtr);
```

Comment:
default value would be useful
```suggestion
    void reset(T * newPtr = nullptr);
```